File: code/design-patterns/behavioral/state/go/state_usage.go
package main

// Example usage
func main() {
	vendingMachine := NewVendingMachine()

	vendingMachine.AddItem()
	vendingMachine.SelectItem()
	vendingMachine.InsertMoney()
	vendingMachine.DispenseItem()
	vendingMachine.AddItem()
}

File: code/design-patterns/structural/state/python/state_usage.py
from state import VendingMachine

# Example usage
vendingMachine = VendingMachine()

vendingMachine.addItem()
print(vendingMachine.currentState)
vendingMachine.selectItem()
print(vendingMachine.currentState)
vendingMachine.insertMoney()
print(vendingMachine.currentState)
vendingMachine.dispenseItem()
print(vendingMachine.currentState)
vendingMachine.addItem()
